gensim/models/word2vec.py

```diff
@@ -432,6 +432,8 @@ def __init__(
 
         """
 
+        self.load = call_on_class_only
```

Comment:
@piskvorky, that is what I did at the beginning. Apart from the points @gojomo raises, there are some issues with the way `types.MethodTypes` is handled in Python versions 2 and 3.